### PR TITLE
Merchant page

### PR DESCRIPTION
**Pull request by xmrdsc**
This PR introduces a dedicated merchant page. Current iteration of the merchant features present on the receive page are inadequate to be used as a point of sales system.

The changes in this PR will make it easier for merchants to accept Monero. Run the application in full-screen mode, set the amount to be received and customers scan the QR code.

Alternatively you can provide a payment URL.

Some notes:

- Introduces a menu under Receive called Merchant.
- Optimized for running in full-screen.
- Transaction history shows a 'X time ago' indication. i.e: "the transaction happened 5 minutes ago"
- Incoming transactions are monitored via tx pool.
- For now, clicking on transactions takes you to the History page.
- In the future merchants could optionally specify their to received amount in USD when #1714 is accepted & merged.

[Video](https://i.imgur.com/xk2dlEA.mp4)

### CODE REVIEW

**Comment by xiphon** [comment]
Would like to keep QR Code and Payment URL field on the Receive tab as it is done in the current master.
Using `Merchant` when you want to receive some funds from a friend looks wrong/misleading to me.

**Comment by xmrdsc** [comment]
@xiphon Agreed.

In addition; more feedback from [Reddit](https://www.reddit.com/r/Monero/comments/a4kzf1/feedback_merchant_page_in_the_gui/):

- [ ] If I'm going to display this page to my client, checking "enable sales tracker" should never display my old transactions and amounts! Otherwise my client will know how much the previous client has spent in his dinner.
- [ ] If I'm going to display this page to my client, the disclaimer "it's up to you to accept unconfirmed transactions or not" should appear only the first time this page is opened. My client doesn't need to read this.
- [ ] Amount font should be large.
- [ ] Displaying a Monero logo would be nice.
- [ ] What's preventing the costumer from "tab-ing" back to the regular GUI and exposing the main address, full amount

Will fix these.

**Comment by dEBRUYNE-1** [comment]
Perhaps we could add the QR code to both the receive page and the merchant page?

**Comment by xmrdsc** [comment]
@dEBRUYNE-1 

https://i.imgur.com/lMLbkbK.png

**Comment by xmrdsc** [comment]
Hiding balance:


Hiding amounts:

**Comment by xmrdsc** [comment]
To sum up the changes above:

- Made most text selectable on the merchant page
- Ability to hide the incoming amounts
- Receive page still has QR/Payment URL under 'Advanced Options'
- 'Settings->Layout received a new persistent setting 'Hide Balance'.
- Increased font size for the amount to receive input
- Added Monero logo on the merchant page

**Comment by xiphon** [comment]
> Toggle hiding balance via Settings->Layout->Hide balance 

Please submit this in a separate PR

**Comment by dEBRUYNE-1** [comment]
@xmrdsc - I think we should remove the Payment URL and Amount box from the `Receive` page, as I can tell from experience (of helping / assisting users) that it will merely confuse the normal user.

**Comment by selsta** [comment]
@dEBRUYNE-1 I see uses for it and it’s under Advanced options anyway, let’s keep the super simple  things for the simple mode :)

**Comment by xmrdsc** [comment]
@selsta @dEBRUYNE-1 Middle ground: remove `Amount to receive` - keep QR & Payment URL. Yeah?

**Comment by xmrdsc** [comment]
Rebased to latest master.

- Fixed QR code save dialog @selsta 
- Moved commit to hide balance to separate PR #1802 @xiphon 
- Removed 'Amount to receive' from Receive page; [see picture](https://i.imgur.com/eeuod89.png) @dEBRUYNE-1

**Comment by selsta** [comment]
I think ideally the receive page would look like this: 
screenshot 2018-12-13 at 11 15 52
Just Addresses + QR code with no advanced options. The payment URL would be hidden inside the QR code (#1810) and the help text would be updated.

**Comment by xmrdsc** [comment]
As the discussion revolves around how the receive page should look like, I'm in favor of reviewing the merchant page and get that in first.

**Comment by xmrdsc** [comment]
Rebased.

**Comment by xiphon** [comment]
Please squash

**Comment by xmrdsc** [comment]
I'd say commit log has already been carefully curated.

**Comment by erciccione** [comment]
I agree. With large changes like this one i' d prever more verbosity, therefore, more commits

**Comment by xiphon** [comment]
> I agree. With large changes like this one i' d prever more verbosity, therefore, more commits

This is not the case. Each commit must be self-consistent.

**Comment by xmrdsc** [comment]
@xiphon I rebased & squashed:

- Add merchant page and components
- Update qml.qrc and .pro

**Comment by mmbyday** [comment]
+1 for squashing all
+1 for self-consistent, finished PRs.

**Comment by xiphon** [comment]
https://github.com/monero-project/monero-gui/pull/1793#pullrequestreview-186121274 was just an example.

Okay, the next one.

`Prepare MiddlePanel for views with no padding/margins and custom background colors` commit is not self-consistent. It uses `moneroGrey` that doesn't exist at this point.

* Prepare MiddlePanel for views with no padding/margins and custom background colors
* Add Monero grey to style

**Comment by xmrdsc** [comment]
Squashed all.

File: components/TitleBar.qml [hunk 1]
```diff
@@ -56,6 +56,9 @@ Rectangle {
     property bool showMoneroLogo: false
     property bool small: false
     property alias titleBarGradientImageOpacity: titleBarGradientImage.opacity
+    property bool orange: false
+    property string buttonHoverColor: "#262626"
+    property string buttonHoverColorOrange: "#44FFFFFF"
```
Comment (xiphon) [inline components/TitleBar.qml]:
```suggestion
    property string buttonHoverColorOrange: "#44FFFF"
```

File: pages/merchant/Merchant.qml [hunk 2]
```diff
@@ -0,0 +1,697 @@
...
+            }
+
+            DropShadow {
+                anchors.fill: source
+                cached: true
+                horizontalOffset: 3
+                verticalOffset: 3
+                radius: 8.0
+                samples: 16
+                color: "#30000000"
```
Comment (xiphon) [inline pages/merchant/Merchant.qml]: Transparent?

File: pages/merchant/Merchant.qml [hunk 3]
```diff
@@ -0,0 +1,697 @@
...
+            }
+
+            DropShadow {
+                anchors.fill: source
+                cached: true
+                horizontalOffset: 3
+                verticalOffset: 3
+                radius: 8.0
+                samples: 16
+                color: "#20000000"
```
Comment (xiphon) [inline pages/merchant/Merchant.qml]: Same

File: pages/merchant/Merchant.qml [hunk 4]
```diff
@@ -0,0 +1,697 @@
...
+                width: (parent.width - qrImg.width) - (50 * scaleRatio)
+                height: 32 * scaleRatio
+
+                Text {
+                    anchors.verticalCenter: parent.verticalCenter
+                    anchors.horizontalCenter: parent.horizontalCenter
+                    font.pixelSize: 12 * scaleRatio
+                    font.bold: false
+                    color: "white"
+                    text: "<style type='text/css'>a {text-decoration: none; color: #FF6C3C; font-size: 12px;}</style>Currently selected address: " + addressLabel + " <a href='#'>(Change)</a>"
```
Comment (xiphon) [inline pages/merchant/Merchant.qml]: qsTr and place the tags outside

File: pages/merchant/Merchant.qml [hunk 5]
```diff
@@ -0,0 +1,697 @@
...
+                    textFormat: Text.RichText
+
+                    MouseArea {
+                        anchors.fill: parent
+                        hoverEnabled: true
+                        cursorShape: Qt.PointingHandCursor
+                        onClicked:appWindow.showPageRequest("Receive")
```
Comment (xiphon) [inline pages/merchant/Merchant.qml]:
```suggestion
                        onClicked: appWindow.showPageRequest("Receive")
```

File: pages/TxKey.qml [hunk 6]
```diff
@@ -45,7 +45,7 @@ Rectangle {
     /* main layout */
     ColumnLayout {
         id: mainLayout
-        anchors.margins: 40 * scaleRatio
+        anchors.margins: (isMobile)? 17 : 20
```
Comment (xiphon) [inline pages/TxKey.qml]: scaleRatio

File: pages/merchant/MerchantCheckbox.qml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    DropShadow {
+        anchors.fill: source
+        cached: true
+        horizontalOffset: 3
+        verticalOffset: 3
+        radius: 8.0
+        samples: 16
+        color: "#20000000"
```
Comment (mmbyday) [inline pages/merchant/MerchantCheckbox.qml]: Hex code looks too long.

File: pages/merchant/MerchantTrackingList.qml [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+
+                    TextEdit {
+                        anchors.horizontalCenter: parent.horizontalCenter
+                        anchors.verticalCenter: parent.verticalCenter
+                        font.pixelSize: 12 * scaleRatio
+                        font.bold: false
+                        color: "#a8a8a8"
+                        text: {
+                            if(in_txpool){
+                                return qsTr("Awaiting in txpool")
```
Comment (mmbyday) [inline pages/merchant/MerchantTrackingList.qml]:
```suggestion
                                return qsTr("Awaiting in txpool") + translationManager.emptyString;
```

File: components/TitleBar.qml [hunk 9]
```diff
@@ -70,30 +73,48 @@ Rectangle {
...
         }
+
+        Image {
+            visible: !isMobile && showMoneroLogo && titleBar.orange
+            anchors.left: parent.left
+            anchors.top: parent.top
+            anchors.topMargin: 11
+            width: 132
+            height: 22
+            source: "../images/moneroLogo_white.png"
```
Comment (mmbyday) [inline components/TitleBar.qml]: have any more of those  2x images? :-)

File: components/TitleBar.qml [hunk 10]
```diff
@@ -130,7 +151,13 @@ Rectangle {
             hoverEnabled: true
             anchors.fill: parent
             cursorShape: Qt.PointingHandCursor
-            onEntered: goToBasicVersionButton.color = "#262626";
+            onEntered: {
```
Comment (mmbyday) [inline components/TitleBar.qml]:
```suggestion
            onEntered: { goToBasicVersionButton.color = titleBar.orange ? titleBar.buttonHoverColorOrange : titleBar.buttonHoverColor }
```

File: pages/merchant/Merchant.qml [hunk 11]
```diff
@@ -0,0 +1,697 @@
...
+                        color: "#d9d9d9"
+                    }
+
+                    MerchantTrackingList {
+                        Layout.fillWidth: true
+                        Layout.preferredHeight: 400 * scaleRatio
+                        model: trackingModel
+                        message: {
+                            if(!root.enableTracking){
+                                return qsTr(
```
[the remark this answers is not in the document]
Comment (xiphon) [reply pages/merchant/Merchant.qml]: @erciccione what is the best approach from the translator's point of view? I believe strings containing tags may confuse some translators.

**Comment by xiphon** [comment]
Let's remove `Amount to receive (USD)` for now and add it after #1714 gets merged.

File: components/TitleBar.qml (hunk 1 above)
Comment (xmrdsc) [reply components/TitleBar.qml]: `#44FFFFFF` is actually correct, try comparing the two.

File: pages/merchant/Merchant.qml (hunk 11 above)
[the remark this answers is not in the document]
Comment (xmrdsc) [reply pages/merchant/Merchant.qml]: Leaving it as is for now, however I do indeed think it's better to separate the two. Should be part of a different PR, mixed tags/translatable strings are scattered around our codebase.

File: pages/merchant/Merchant.qml (hunk 2 above)
Comment (xmrdsc) [reply pages/merchant/Merchant.qml]: Slightly transparent. Try `#FF000000` and compare.

File: pages/merchant/Merchant.qml (hunk 3 above)
Comment (xmrdsc) [reply pages/merchant/Merchant.qml]: Works as intended.

File: pages/merchant/Merchant.qml (hunk 4 above)
Comment (xmrdsc) [reply pages/merchant/Merchant.qml]: See previous reply regarding mixing of tags/translatable strings.

File: pages/merchant/Merchant.qml (hunk 5 above)
Comment (xmrdsc) [reply pages/merchant/Merchant.qml]: Fixed.

File: pages/merchant/MerchantCheckbox.qml (hunk 7 above)
Comment (xmrdsc) [reply pages/merchant/MerchantCheckbox.qml]: As intended, including transparency.

File: pages/TxKey.qml (hunk 6 above)
Comment (xmrdsc) [reply pages/TxKey.qml]: Fixed;

```
./pages/Transfer.qml
./pages/Sign.qml
./pages/SharedRingDB.qml
./pages/TxKey.qml
./pages/settings/SettingsWallet.qml
./pages/settings/SettingsLog.qml
./pages/settings/SettingsNode.qml
./pages/settings/SettingsInfo.qml
./pages/settings/SettingsLayout.qml
./pages/Receive.qml
./pages/AddressBook.qml
./pages/Keys.qml
./pages/Mining.qml

```

File: components/TitleBar.qml (hunk 10 above)
Comment (xmrdsc) [reply components/TitleBar.qml]: Fixed.

**Comment by xmrdsc** [comment]
@xiphon 

> Let's remove Amount to receive (USD) for now and add it after #1714 gets merged.

File: pages/merchant/MerchantCheckbox.qml (hunk 7 above)
Comment (xmrdsc) [reply pages/merchant/MerchantCheckbox.qml]: http://doc.qt.io/qt-5/qml-color.html

File: pages/merchant/Merchant.qml (hunk 3 above)
Comment (xmrdsc) [reply pages/merchant/Merchant.qml]: http://doc.qt.io/qt-5/qml-color.html

File: components/TitleBar.qml (hunk 1 above)
Comment (xmrdsc) [reply components/TitleBar.qml]: http://doc.qt.io/qt-5/qml-color.html

File: components/TitleBar.qml (hunk 9 above)
Comment (xmrdsc) [reply components/TitleBar.qml]: I do not :-)

File: pages/merchant/MerchantTrackingList.qml (hunk 8 above)
Comment (xmrdsc) [reply pages/merchant/MerchantTrackingList.qml]: Fixed

File: pages/merchant/Merchant.qml (hunk 2 above)
Comment (xiphon) [reply pages/merchant/Merchant.qml]: :ok_hand:

File: pages/merchant/Merchant.qml (hunk 3 above)
Comment (xiphon) [reply pages/merchant/Merchant.qml]: :ok_hand:

File: pages/merchant/Merchant.qml (hunk 4 above)
Comment (xiphon) [reply pages/merchant/Merchant.qml]: :ok_hand:

**Pull request merged**